src/encoded/auth0.py

```
@@ -96,6 +101,56 @@ def forget(self, request):
         return []
 
 
+@view_config(context=User.Collection, request_method='POST', permission=NO_PERMISSION_REQUIRED, name='sign-up')
+def signup(context, request):
+    """
+    Create new user.
+
+    :param request: Pyramid request object
+    """
+    domain = 'encode.auth0.com'
+    access_token = request.json.get('accessToken', '')
+    if not access_token:
+        raise HTTPBadRequest(explanation='Access token required.')
+    url = "https://{domain}/userinfo?access_token={access_token}".format(domain=domain, access_token=access_token)
+    user_data_request = requests.get(url)
+    if user_data_request.status_code != 200:
+        raise HTTPBadRequest(explanation='Could not get user data')
+    user_data = user_data_request.json()
+    if user_data['email_verified'] is not True:
+        raise HTTPBadRequest(explanation='Unverified e-mail')
+    first_name, last_name = _get_first_and_last_names_from_name(
+        user_data.get('name', ''),
+        user_data['email']
+    )
+    user_info = {
+        'email': user_data['email'],
+        'first_name': user_data.get('given_name', '') or user_data.get('first_name', '') or first_name,
+        'last_name': user_data.get('family_name', '') or user_data.get('last_name', '') or last_name,
+    }
+    validate_request(context.type_info.schema, request, user_info)
+    if request.errors:
+        raise HTTPUnprocessableEntity(json={'errors':request.errors}, content_type='application/json')
```

This should raise a validation error.